controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1186,16 +1191,162 @@ bool master_thread::handle_cmdu_1905_higher_layer_data_message(Socket *sd,
+                LOG(ERROR) << "addClass wfa_map::tlvClientAssociationEvent failed";
+                return false;
+            }
+            break;
+        case int(ieee1905_1::eTlvType::TLV_VENDOR_SPECIFIC):
+            LOG(DEBUG) << "Found TLV_VENDOR_SPECIFIC TLV";
+
+            if (!message_com::parse_intel_vs_message(cmdu_rx)) {
+                LOG(DEBUG) << "Failed to parse intel vs message (not Intel?)";
+                continue;
+            }
+
+            vs_tlv = cmdu_rx.addClass<beerocks_message::tlvVsClientAssociationEvent>();
+            if (!vs_tlv) {
+                LOG(ERROR) << "addClass wfa_map::tlvVsClientAssociationEvent failed";
+                return false;
+            }
+            break;
+        default:
+            LOG(ERROR) << "Unexpected TLV. Ignore CMDU.";
+            return false;
+        }
+    }
+
+    if (!client_association_event_tlv) {
+        LOG(ERROR) << "wfa_map::tlvClientAssociationEvent not found";
+        return false;
+    }
+
+    auto &client_mac    = client_association_event_tlv->client_mac();
+    auto client_mac_str = network_utils::mac_to_string(client_mac);
+
+    auto &bssid    = client_association_event_tlv->bssid();
+    auto bssid_str = network_utils::mac_to_string(bssid);
+
+    auto &association_event = client_association_event_tlv->association_event();
+
+    LOG(INFO) << "client "
+              << (association_event == wfa_map::tlvClientAssociationEvent::CLIENT_HAS_JOINED_THE_BSS
+                      ? "connected"
+                      : "disconnected")
+              << ", client_mac=" << client_mac_str << ", bssid=" << bssid_str;
+
+    if (association_event == wfa_map::tlvClientAssociationEvent::CLIENT_HAS_JOINED_THE_BSS) {
+        // client connected
+
+        //add or update node parent
+        database.add_node(client_mac_str, bssid_str);
+
+        LOG(INFO) << "client connected, mac=" << client_mac_str << ", bssid=" << bssid_str;
+
+        database.set_node_channel_bw(client_mac_str, database.get_node_channel(bssid_str),
+                                     database.get_node_bw(bssid_str),
+                                     database.get_node_channel_ext_above_secondary(bssid_str), 0,
+                                     database.get_hostap_vht_center_frequency(bssid_str));
+
+        database.clear_node_cross_rssi(client_mac_str);
+        database.clear_node_stats_info(client_mac_str);
+
+        if (!(database.get_node_type(client_mac_str) == beerocks::TYPE_IRE_BACKHAUL &&
+              database.get_node_handoff_flag(client_mac_str))) {
+            // The node is not an IRE in handoff
+            database.set_node_type(client_mac_str, beerocks::TYPE_CLIENT);
+        }
+
+        database.set_node_backhaul_iface_type(client_mac_str,
+                                              beerocks::IFACE_TYPE_WIFI_UNSPECIFIED);
+
+        if (vs_tlv) {
+            database.set_node_vap_id(client_mac_str, vs_tlv->vap_id());
```

Comment:
Shouldn't we clear the vap ID if there is no vs_tlv?

The station capabilities we probably want to keep (when roaming from an Intel to a non-Intel AP, the capabilities will not have changed, but we don't know them from the non-Intel AP).

Reply:
I'm not sure, Basically, whats important is the BSSID of the vap (which we already get and update and implies the vap_id) so we can use it.
I think it is better to remove vap_id "term" from the code and use bssid.
But it needs to be in a different task/PR.
Clearing/Setting the vap_id here is no different than not updating it, by how it affects the system.